pkg/registry/pod/etcd/etcd.go

```diff
@@ -218,6 +220,38 @@ type LogREST struct {
 	kubeletConn client.ConnectionInfoGetter
 }
 
+// Max length read from the body of a log location which returns error status
+const (
+	maxReadLength = 1000
+)
+
+// A log location response checker to transform the error.
+type LogLocationResponseChecker struct {
```

Comment:
In a separate issue we need to make the non etcd parts of the Pod REST implementation less coupled to etcd by defining the interfaces between them and the store more cleanly and moving them out of `pkg/registry/pod/etcd` - for now just not adding more stuff there is good.